packages/@aws-cdk/aws-synthetics/lib/canary-base.ts

```diff
@@ -0,0 +1,11 @@
+import * as cdk from '@aws-cdk/core';
+
+export interface ICanary extends cdk.IResource {
+  // Should have all the properties and method signatures associated with a canary
+}
```

Comment:
Please refrain from future-proofing. It's easy to add bases (more interfaces, more abstract base classes, ...), but it's a breaking change to remove them.